spaceship-prompt/spaceship-prompt
# Adding support for aws-vault sessions in aws section

### What has been done
- `aws-vault` creates temporary credentials and correctly unsets `AWS_PROFILE` to prevent tools from ignoring the environment variables containing the temporary keys. However, `aws-vault` helpfully provides an `AWS_VAULT` that contains the name of the profile it used to create the session.
- New variable `SPACESHIP_AWS_PROFILE` used in section gets value of `AWS_PROFILE` if available, falling back to the value of `AWS_VAULT` if available.

### How to test
- Verify the `aws` section behaves as before with `AWS_PROFILE` set and unset.
- Install `aws-vault` if necessary and add a profile.
- Verify that Spaceship shows the correct profile name after creating a timed session subshell with `aws-vault exec -t 1h <PROFILE_NAME>`


Close #663
